pkg/broker/receiver.go

```diff
@@ -237,37 +237,84 @@ func (r *Receiver) getTrigger(ctx context.Context, ref path.NamespacedNameUID) (
 }
 
 // shouldSendMessage determines whether message 'm' should be sent based on the triggerSpec 'ts'.
-// Currently it supports exact matching on type and/or source of events.
+// Currently it supports exact matching on event context attributes.
+// If no filter strategy is present, shouldSendMessage returns true.
 func (r *Receiver) shouldSendMessage(ctx context.Context, ts *eventingv1alpha1.TriggerSpec, event *cloudevents.Event) bool {
-	if ts.Filter == nil || ts.Filter.SourceAndType == nil {
+	if ts.Filter == nil {
```

Comment:
Based on the updated method comment, shouldn't this return true?

Reply:
Updated the comment. The webhook is creating an empty filter, with no strategy. If that doesn't happen, then we return false. If the filter is specified with no strategy (either DeprecatedSourceAndType or Attributes), then we return true.